src/runtime_src/core/pcie/windows/shim.cpp

```diff
@@ -1085,6 +1099,23 @@ xclUnmgdPread(xclDeviceHandle handle, unsigned int flags, void *buf, size_t coun
   return shim->unmgd_pread(flags, buf, count, offset);
 }
 
+ssize_t
+xclMgdPwrite(xclDeviceHandle handle, unsigned int flags, const void *buf, size_t count, uint64_t offset)
+{
+    xrt_core::message::
+        send(xrt_core::message::severity_level::XRT_DEBUG, "XRT", "xclMgdPwrite()");
+    auto shim = get_shim_object(handle);
+    return shim->mgd_pwrite(flags, buf, count, offset);
+}
+
+ssize_t
+xclMgdPread(xclDeviceHandle handle, unsigned int flags, void *buf, size_t count, uint64_t offset)
```

Comment:
If we are changing the xclMgdPwrite to xclWriteBO, then we should change this to xclReadBO.